src/w2grid.js

```diff
@@ -701,7 +702,8 @@
             // process date fields
             obj.selectionSave();
             obj.prepareData();
-            obj.reset();
+            if (!noReset)
+                obj.reset();
```

Comment:
Ok, sorry for the bad description. The sort is not really messed up, it's that the user experience is really bad, because obj.reset() scroll the view back to the top. So if you have grid.html with a reversed sort, and a lot of lines like that:

![screen shot 2016-01-02 at 22 57 56](https://cloud.githubusercontent.com/assets/7224918/12076285/24db5a94-b1a5-11e5-8e80-0ddc41984d90.png)

When you click to open the node (2) at row 4963, the view scrolls back to the top (row 1) which is painful. I've the same problem without the tree view where I've a websocket that slowly calls add() to add entries to the end of a grid, 100 new records every second up to 50k records. If the add() method calls reset(), the view scrolls back to row 1 every second too, which means you cannot scroll and use the grid while loading... This is fixed by this commit.